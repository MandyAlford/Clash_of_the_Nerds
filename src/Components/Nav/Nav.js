import React from 'react';
import './Nav.scss';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import small from '../../assets/logo-small.png';


const Nav = (props) => {
  console.log(props.user[0].name);
  return (
    <section className='nav-bar'>
      <h1 className='user-name-nav'>Welcome {props.user[0].name}</h1>
      <img src={ small } className='small-title-logo' alt='clash of the nerds logo but small' />
      <Link to='/'><button type='button' className='home-btn'>Quit</button></Link>
    </section>
  )
}

export const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps)(Nav)
