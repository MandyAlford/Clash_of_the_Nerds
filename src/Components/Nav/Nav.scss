.nav-bar {
  background-color: #662324;
  height: 60px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  position: relative;
  z-index: 150;
}

.user-name-nav {
  color: #FFF;
  font-size: 25px;
  float: left;
  text-decoration: none;
}

.home-btn {
  color: #FFF;
  height: 35px;
  font-size: 20px;
  margin-right: 50px;
  width: 100px;
  color: #662324;
  border-radius: 5px;
}

.user-info {
  display: flex;
  flex-direction: row;
  align-items: center;
  text-decoration: none;
}

.small-title-logo {
  height: 100px;
}

.user-icon {
  height: 60px;
  margin-left: 30px;
}
