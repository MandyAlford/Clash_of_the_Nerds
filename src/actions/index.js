export const addUser = (user) => ({
    type: 'ADD_USER',
    user
})

export const addTrivia = (trivia) => ({
    type: 'ADD_TRIVIA',
    trivia
})

export const addCorrectGuess = (guess) => ({
  type: 'ADD_CORRECT_GUESS',
  guess
})

export const addIncorrectGuess = (guess) => ({
  type: 'ADD_INCORRECT_GUESS',
  guess
})

export const resetAnswers = () => ({
  type: 'RESET_STORE_ANSWERS'
})

export const saveUsersTrivia = (trivia) => ({
  type: 'SAVE_TRIVIA',
  trivia
})
