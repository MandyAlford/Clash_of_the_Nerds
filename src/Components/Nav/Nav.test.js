import React from 'react';
import { shallow } from 'enzyme';
import { Nav, mapStateToProps, mapDispatchToProps } from './Nav';
import { resetAnswers } from '../../actions';


describe('Nav', () => {
  let wrapper, mockUser;

  beforeEach(() => {
    mockUser = {
      name: 'Jessica Rabbit',
      team: '19',
      level: 'hard'
    }
    wrapper = shallow(<Nav
      user={[mockUser]}
    />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should update state when mapStateToProps is called', () => {
    let mockState = {
      user: {
        name: 'Jessica Rabbit',
        team: '19',
        level: 'hard'
      },
      trivia: [{}]
    }
    let expected = {
      user: {
        name: 'Jessica Rabbit',
        team: '19',
        level: 'hard'
      }
    }
    const result = mapStateToProps(mockState)
    expect(result).toEqual(expected)
  })

  it('should call dispatch when loadIncorrectGuessToStore is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = resetAnswers()
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.loadIncorrectGuessToStore()
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

})
