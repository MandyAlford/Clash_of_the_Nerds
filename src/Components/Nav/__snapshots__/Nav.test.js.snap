// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
<section
  className="nav-bar"
>
  <Link
    to="/account"
  >
    <div
      className="user-info"
    >
      <img
        alt="small spartan man for user account access"
        className="user-icon"
        src="spartan-icon.png"
      />
      <h1
        className="user-name-nav"
      >
        Welcome 
        Jessica Rabbit
      </h1>
    </div>
  </Link>
  <img
    alt="clash of the nerds logo but small"
    className="small-title-logo"
    src="logo-small.png"
  />
  <Link
    to="/"
  >
    <button
      className="home-btn"
      onClick={[Function]}
      type="button"
    >
      Quit to Home
    </button>
  </Link>
</section>
`;
